File: src/sum_squares_perf.go
package main

import (
    "encoding/json"
    "fmt"
    "math/big"
    "time"
)

type Output struct {
    Result         *big.Int `json:"result"`
    ElapsedTimeMs  float64  `json:"elapsed_time_ms"`
}

// sumOfSquares calculates the sum of squares from 1 to n using big.Int.
func sumOfSquares(n int) *big.Int {
    sum := big.NewInt(0)
    for i := 1; i <= n; i++ {
        square := big.NewInt(int64(i))
        square.Mul(square, square)
        sum.Add(sum, square)
    }
    return sum
}

func main() {
    startTime := time.Now()
    result := sumOfSquares(10000000)
    elapsedTime := time.Since(startTime).Seconds() * 1000 // in milliseconds

    output := Output{
        Result:        result,
        ElapsedTimeMs: elapsedTime,
    }

    jsonOutput, err := json.Marshal(output)
    if err != nil {
        fmt.Println("Error marshalling to JSON:", err)
        return
    }

    fmt.Println(string(jsonOutput))
}

File: src/sum_squares_perf.py
import json
import time


def sum_of_squares(n):
    return sum(i * i for i in range(1, n + 1))


if __name__ == "__main__":
    start_time = time.time()
    result = sum_of_squares(10_000_000)
    end_time = time.time()
    elapsed_time_ms = (end_time - start_time) * 1000

    # Output the result and the time as JSON
    output = {"result": result, "elapsed_time_ms": elapsed_time_ms}
    print(json.dumps(output))
